### ISSUE TITLE
Anlegen eines Kontaktpersons mit der API

### ISSUE THREAD

**Issue by Torchok19081986** [selectline-software/selectline-api#252]
Hallo, ich habe versucht einige Kontaktpersonen des Kunden anzulegen. Leider konnte ich das nicht. Abruf der Kontaktpersonen by Nummer von Kunden läuft einwandfrei , Route /Customers/{customerNumber}/ContactPersons, aber diese in weiteren Mandanten anzulegen, konnte ich  nicht machen. 

Es kommt weder Fehlermeldung, noch die Personen werden übertragen. 
Ist da richtig, dass der Parentnumber ein Number der Kunden zu welchem Kontaktperson gehört und Numer in ContactPersons Number des ContactPersons ? 
Im Wawi unter Kontaktpersonen sehe ich keine Parentnumer. Die Spalten Kunde oder Lieferant  sind leer, obwohl die API in Parentnumer gleiche Number liefert wie das abgerufene Kunde.
Wie kann ich die Kontaktpersonen zu einen Kunden anlegen? Muss ich evtl. mehrere Schritte bzw. mehrere Routen ablaufen?

**Comment by MatthiasGuse** [comment]
Hallo,

leider kann ich Ihrer Beschreibung nicht entnehmen, was genau Ihr Problem ist. Evtl. zeigen Sie mal ein Beispiel, welches bei Ihnen fehlschlägt. Sonst besteht die Möglichkeit den kostenpflichtigen Support zu nutzen.
Das Anlegen der Kontaktpersonen erfolgt über die genannte Route POST /Customers/{customerNumber}/ContactPersons. Wichtig ist die korrekte Angabe der Kundennummer in der URL. Das muss eine vorhandene Nummer eines Kunden sein. Sonst reicht es meines Wissens nach, die gewünschten Felder anzugeben.

Viele Grüße

**Comment by Torchok19081986** [comment]
morgen, danke für die Info. 
```csharp

 var kontaktpersonListe = customerApi.CustomerReadContactPersonsAsync(kunde.Number, null, null, null, null, "Number", "ParentNumber EQ " + kontaktperson.ParentNumber);
````

und dann wenn anzahl ist gleich 0 

```csharp
 ApiResponse<object> response = customerApi.CustomerCreateContactPersonAsyncWithHttpInfo(kunde.Number, kontaktperson);

```

Hier erscheint weder die Fehlermedung, noch die Meldung von der API, dass die Kontaktperson angelegt wurde. Response ist nichts.

**Comment by MatthiasGuse** [comment]
Sorry, damit können wir nichts anfangen.

Um hier weiterzukommen, benötigen wir ein Beispiel, wie wir das beschriebene Fehlverhalten per Swagger oder Postman nachvollziehen können.

Grüße

**Comment by Torchok19081986** [comment]
morgen, ich konnte jetzt zumind. eine Fehlemeldung provozieren. 
Im Swagger anmelden beide Mobile API Benutzer und anschlißend authorisiert --> erfolgreich.
mit o.g. Route die ContactPerson by Nummer lesen und Ergebnis beim 2. Benutzer ins Post Method kopieren -> bis jetzt alles ok.
Wenn man aber versucht, Post auszuführen, erscheint bei mir im Testumgebung --> MetaDaten WaWiUserToken is  invaid and required max length of 2 chars.

**Comment by Micha-Richter** [comment]
Ist denn das Ergebnis-Model eines Elements von CustomerReadContactPersonsAsync(...) vom Typ identisch zum Model des Parameters von CustomerCreateContactPersonAsyncWithHttpInfo(...)?

**Comment by Torchok19081986** [comment]
Ja, Ergebnis ist der Model von Type SelectlineErp...ContactPerson.

**Comment by Micha-Richter** [comment]
Versuch mal das Post ohne MetaData.

**Comment by Torchok19081986** [comment]
ok, probiere ich jetzt gleich aus.

**Comment by Micha-Richter** [comment]
@MatthiasGuse 
Die Fehlermeldungen "MetaDaten WaWiUserToken is invalid" und "required max length of 2 chars" sollten aber trotzdem korrigiert werden!? Es gibt durchaus nur 1stellige Wawi-Kürzel 🤷‍♂️

**Comment by Torchok19081986** [comment]
Ok. vielen Dank. Warte auf neue Swagger Beschreibung in der nächsten Version. 👍👍

**Comment by Micha-Richter** [comment]
> Wie kann ich die Metadaten aus dem API ausschließen, wenn es überhaupt möglich ist?

Du musst die Metadaten aus dem Parameter-Objekt des CustomerCreateContactPersonAsyncWithHttpInfo-Aufrufs entfernen.
